src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -452,7 +468,10 @@ private BoundExpression TransformCompoundAssignmentLHS(BoundExpression originalL
                         var propertyAccess = (BoundPropertyAccess)originalLHS;
                         if (propertyAccess.PropertySymbol.RefKind == RefKind.None)
                         {
-                            return TransformPropertyAccess(propertyAccess, stores, temps);
+                            // CONSIDER: this is a temporary object that will be rewritten away before this lowering completes.
+                            // Mitigation: this will only produce short-lived garbage for compound assignments and increments/decrements of properties.
+                            return propertyAccess.Update(TransformPropertyOrEventReceiver(propertyAccess.PropertySymbol, propertyAccess.ReceiverOpt, stores, temps),
+                                                         propertyAccess.PropertySymbol, propertyAccess.ResultKind, propertyAccess.Type);
```

Comment:
Minor point: The corresponding comment for fields does not include `CONSIDER: ... Mitigation: ...`. Perhaps drop that here and use the simpler comment from fields here instead.